pkg/server/status.go

```diff
@@ -2542,3 +2545,39 @@ func (s *statusServer) JobStatus(
 
 	return &serverpb.JobStatusResponse{Job: res}, nil
 }
+
+// GenerateJoinToken generates a new ephemeral join token. For use by the sql
+// subsystem directly. The response is a base64 marshalled form of the join token
+// that can be shared to new nodes that want to join this cluster.
+func (s *statusServer) GenerateJoinToken(ctx context.Context) (string, error) {
+	if !sql.FeatureTLSAutoJoinEnabled.Get(&s.st.SV) {
+		return "", errors.New("join token generation disabled")
+	}
+
+	var jt JoinToken
+	jt.TokenID = uuid.MakeV4()
+	r := rand.New(rand.NewSource(timeutil.Now().UnixNano()))
+	jt.SharedSecret = randutil.RandBytes(r, joinTokenSecretLen)
+
+	certLocator := security.MakeCertsLocator(s.cfg.SSLCertsDir)
+	if ok, err := certLocator.HasNodeCert(); err != nil || !ok {
+		return "", errors.New("cannot generate join token on node without CA cert")
+	}
+
+	// TODO(bilal): This is definitely wrong.
+	f, err := os.Open(certLocator.CACertPath())
```

Comment:
@aaron-crl This likely calls for a "get me the CA cert" function inside `pkg/security, right? Is there an obvious way to get hold of a certificate manager to get this?